pkg/finder/crifinder.go

```diff
@@ -0,0 +1,334 @@
+package finder
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+	"google.golang.org/grpc"
+
+	"k8s.io/api/core/v1"
+
+	pb "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"k8s.io/kubernetes/pkg/kubelet/util"
+)
+
+const (
+	defaultTimeout     = 5 * time.Second
+	ns                 = "resource-topology-exporter"
+	CGroupCPUSetPrefix = "fs/cgroup/cpuset"
+	CGroupCPUsetSuffix = "cpuset.cpus"
+	ContainerdRuntime  = "containerd"
+	CRIORuntime        = "cri-o"
+)
+
+type CGroupPathTranslator func(sysfs, cgroupPath string) string
+type criFinder struct {
+	args             Args
+	conn             *grpc.ClientConn
+	client           pb.RuntimeServiceClient
+	cgroupTranslate  CGroupPathTranslator
+	NodeResourceData *NodeResources
+}
+
+type ContainerInfo struct {
+	sandboxID      string              `json:"sandboxID"`
+	Pid            uint32              `json:"pid"`
+	Removing       bool                `json:"removing"`
+	SnapshotKey    string              `json:"snapshotKey"`
+	Snapshotter    string              `json:"snapshotter"`
+	RuntimeType    string              `json:"runtimeType"`
+	RuntimeOptions interface{}         `json:"runtimeOptions"`
+	Config         *pb.ContainerConfig `json:"config"`
+	RuntimeSpec    *runtimespec.Spec   `json:"runtimeSpec"`
+}
+
+func NewCRIFinder(args Args, pciResMapConf map[string]string) (Finder, error) {
+	finderInstance := &criFinder{
+		args: args,
+	}
+	var err error
+
+	// At this stage, we only support containerd and cri-o
+	if args.ContainerRuntime == ContainerdRuntime {
+		finderInstance.cgroupTranslate = containerDCGroupPathTranslate
+	} else {
+		//cri-o
+		finderInstance.cgroupTranslate = crioCGroupPathTranslate
+	}
+	finderInstance.NodeResourceData, err = NewNodeResources(finderInstance.args.SysfsRoot, pciResMapConf)
```

Comment:
`NodeResources` code seems to be lost in rebase - if not could you please point me to it?